educational/systems/random_agent.py

```diff
@@ -0,0 +1,370 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, Optional, Tuple, Union
+
+import chex
+import jax
+import jax.numpy as jnp
+from absl import app, flags
+
+from mava import specs as mava_specs
+from mava.utils.environments import debugging_utils, smac_utils
+from mava.utils.loggers import logger_utils
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string("system", "test agent", "What agent is running.")
+flags.DEFINE_string(
+    "base_dir", "~/mava", "Base dir to store experiment data e.g. checkpoints."
+)
+flags.DEFINE_string(
+    "env_name",
+    "simple_spread",
+    "Env name e.g. sinple_spread, 3m or 8m.",
+)
+flags.DEFINE_string(
+    "env_type",
+    "debug",
+    "Env type e.g. debug or smac.",
+)
+flags.DEFINE_string(
+    "env_action_type",
+    "discrete",
+    "Discrete or continous. Only applies for debug env,\
+    other env have their predefined types.",
+)
+flags.DEFINE_integer(
+    "max_total_steps",
+    100000,
+    "Max total steps (across all episodes).",
+)
+
+flags.DEFINE_integer(
+    "num_episodes",
+    None,
+    "Max number of episodes.",
+)
+
+
+@dataclass
+class InitConfig:
+    seed: int = 42
+
+
+@dataclass
+class EnvironmentConfig:
+    env_name: str
+    type: str
+    action_space: str
+    seed: int = 42
+
+
+@dataclass
+class SystemConfig:
+    max_total_steps: int
+    num_episodes: int
+    name: str = "random"
+    seed: int = 42
+
+
+@chex.dataclass(frozen=True, mappable_dataclass=False)
+class RandomSystemState:
+    rng: jnp.ndarray
+
+
+def init(config: InitConfig = InitConfig()) -> InitConfig:
+    """Init system.
+
+    This would handle thing to be done upon system once in the beginning of a run,
+    e.g. set random seeds.
+
+    Args:
+        config : init config.
+    """
+    return config
+
+
+def make_environment(
+    config: EnvironmentConfig,
+) -> Tuple[Any, EnvironmentConfig]:
+    """Init and return environment or wrapper.
+
+    Args:
+        config : env config.
+
+    Returns:
+        env or wrapper.
+    """
+    if config.type == "debug":
+        env, _ = debugging_utils.make_environment(
+            env_name=config.env_name,
+            action_space=config.action_space,
+            random_seed=config.seed,
+        )
+    elif config.type == "smac":
+        env, _ = smac_utils.make_environment(
+            map_name=config.env_name, random_seed=config.seed
+        )
+    return env, config
+
+
+def make_system(
```

Comment:
I still think this is too much boilerplate, but I'll approve here and we can deal with it in #887

Reply:
I accidentally added this here. I have removed it, it should just be in the random agent PR.